consensus/polybft/consensus_runtime.go

```diff
@@ -237,16 +237,16 @@ func (c *consensusRuntime) OnBlockInserted(block *types.Block) {
 	// after the block has been written we reset the txpool so that the old transactions are removed
 	c.config.txPool.ResetWithHeaders(block.Header)
 
-	// handle commitment and proofs creation
-	if err := c.stateSyncManager.PostBlock(&PostBlockRequest{Block: block}); err != nil {
-		c.logger.Error("failed to post block state sync", "err", err)
-	}
-
 	var (
 		epoch = c.epoch
 		err   error
 	)
 
+	// handle commitment and proofs creation
+	if err := c.stateSyncManager.PostBlock(&PostBlockRequest{Block: block}); err != nil {
+		c.logger.Error("failed to post block state sync", "err", err)
+	}
+
 	// TODO - this condition will need to be changed to recognize that either slashing happened
 	// or epoch reached its fixed size
 	if c.isFixedSizeOfEpochMet(block.Header.Number, c.epoch) {
```

Comment:
Would you like to use local variable instead?
```suggestion
	if c.isFixedSizeOfEpochMet(block.Header.Number, epoch) {
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1083/commits/37d13b5aeda743254f04235b861ccd87e76e247f